### PR TITLE
Remove CardSolver from bags

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
This PR removes CardSolver from bags and rely on reducing bag.card operator using bounded quantifiers. 
As a result many unsat regressions timeout and they are skipped. 
This PR also add documentation for skolem ids used by bag.card, bag.fold and set.fold operators.

### CODE REVIEW

File: src/expr/skolem_manager.h [hunk 1]
```diff
@@ -132,13 +132,105 @@ enum class SkolemFunId
    * i = 0, ..., n.
    */
   RE_UNFOLD_POS_COMPONENT,
-  BAGS_CARD_CARDINALITY,
+  /**
+   * An uninterpreted function for bag.fold operator:
```
Comment (ajreynol) [inline src/expr/skolem_manager.h:136]:
```suggestion
   * An uninterpreted function for bag.card operator:
```

File: src/expr/skolem_manager.h [hunk 2]
```diff
@@ -132,13 +132,105 @@ enum class SkolemFunId
...
+   *
+   * - Number of skolem arguments: ``1``
+   *   - ``1:`` an integer index in [0,n]
+   */
   BAGS_CARD_UNION_DISJOINT,
+  /**
+   * An uninterpreted function for bag.fold operator:
+   * To compute (bag.fold f t A), we need to guess the cardinality n of
+   * bag A using a skolem function with BAGS_FOLD_CARD id.
+   *
```
Comment (ajreynol) [inline src/expr/skolem_manager.h]: Missing args.

Comment (ajreynol) [reply src/expr/skolem_manager.h]: Do you mean that the arguments are empty? If so, note 0 arguments.  Same for other operators below.

**Comment by mudathirmahgoub** [comment]
> Can you format this doc similar to how I am doing here: https://github.com/cvc5/cvc5/pull/10393/files#diff-1a7e76b6e2391ee75f078a6322b997fbf34ad89f8ce5aec9ffe03666830f8a8e

Done.

File: src/expr/skolem_manager.h [hunk 3]
```diff
@@ -132,74 +132,187 @@ enum class SkolemFunId
...
+   * counts multiplicities of distinct elements. We call this function
+   * combine of type Int -> Int where:
+   * combine(0) = 0.
+   * combine(i) = m(elements(i), A) + combine(i-1) for 1 <= i <= n.
+   * elements: a skolem function for (bag.fold f t A)
+  *            see BAGS_CARD_ELEMENTS.
+   * n: is the number of distinct elements in A.
+   *
+   * - Number of skolem arguments: ``1``
+   *   - ``1:`` an integer index in [0,n].
```
Comment (ajreynol) [inline src/expr/skolem_manager.h:147]: I don't think this is accurate.

The skolem arguments denote what this skolem is indexed by, which I believe in this case is a bag (the bag to apply cardinality to).

In other words, we are describing the expected `cacheVals` when constructing this function.

File: src/expr/skolem_manager.h [hunk 4]
```diff
@@ -132,74 +132,187 @@ enum class SkolemFunId
...
+   */
   BAGS_CARD_UNION_DISJOINT,
+  /**
+   * An uninterpreted function for bag.fold operator:
+   * To compute (bag.fold f t A), we need to guess the cardinality n of
+   * bag A using a skolem function with BAGS_FOLD_CARD id.
+   */
```
Comment (ajreynol) [inline src/expr/skolem_manager.h:187]: Ditto, this should take the bag to fold as an argument, see https://github.com/cvc5/cvc5/blob/main/src/theory/bags/bag_reduction.cpp#L71C46-L71C60.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/expr/skolem_manager.h (hunk 3 above)
Comment (mudathirmahgoub) [reply src/expr/skolem_manager.h:147]: Ok, I understood it as the arguments to the functions, not the indexing arguments. 
I am missing the context, what is the goal of this documentation?

Comment (ajreynol) [reply src/expr/skolem_manager.h:147]: We are exporting the skolems in proofs and lemmas which users can inspect. Thus, they need full documentation on how they are created.  We potentially will allow users to create these skolems, thus we need to indicate how many arguments they expect.

**Review by ajreynol — APPROVED** [review]
LGTM, can you also go back through old issues and close any that are resolved by this? I imagine there are a number of old issues with `bag.card` where we now timeout.

**Pull request merged**